split_so/models/sale_order.py

```diff
@@ -0,0 +1,55 @@
+from odoo import _, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    # Added field #T00478
+    parent_sale_order_id = fields.Many2one(
+        comodel_name="sale.order", string="Parent sale order"
+    )
+    child_so_count = fields.Integer(
+        string="Child SO", compute="_compute_child_so_count"
+    )
+
+    def _compute_child_so_count(self):
+        """Compute method to calculate number of child sale orders made #T00478"""
+        for count_of_child_so in self:
+            count_of_child_so.child_so_count = self.env["sale.order"].search_count(
+                [("parent_sale_order_id", "=", count_of_child_so.id)]
+            )
+
+    def create_split_so(self):
+        """Function to create child sale order #T00478"""
+        categories = self.order_line.mapped("product_id.categ_id")
+        if len(categories) <= 1:
+            raise ValidationError(_("Cant split, theres only one categories products"))
```

Comment:
Please mention for which category and check the grammar as well
for ex: `raise ValidationError(_("Cant split, there's only one category "%s" products") % category.name)`